src/reducers/sports.js

```diff
@@ -3,38 +3,56 @@ import { RSAA } from 'redux-api-middleware';
 
 import {
   convertArrayToMap,
+  generateTimestamp,
   mergeData,
-  getHeaders,
-  parseResponseData
+  getHeaders
 } from 'utils';
 import { gatewayPath } from 'config';
 
 export const SPORTS_REQUEST = 'SPORTS_REQUEST';
 export const SPORTS_SUCCESS = 'SPORTS_SUCCESS';
 export const SPORTS_FAILED = 'SPORTS_FAILED';
 
-export const fetchSports = timestamp => ({
-  [RSAA]: {
-    endpoint: `${gatewayPath}categories?type=tree&categoryLevel=descendants`,
-    method: 'GET',
-    headers: getHeaders(),
-    types: [
-      SPORTS_REQUEST,
-      {
-        type: SPORTS_SUCCESS,
-        payload: parseResponseData('sports', timestamp)
-      },
-      SPORTS_FAILED
-    ]
-  }
-});
+let sportsTimestampLock = null;
+
+export const checkAliveHandler = newTimestamp => () => {
```

Comment:
U r using the same function in a different ducks hence move it to utils

Reply:
i can't use it multiple times because i need the lock variable to not be closured, in other cases it would be locked on call and would be equal to another timestamp. I was thinking also about the kindof middleware for this, where I import these lock variables